### ISSUE TITLE
"future" parallelism takes a really long time to get going when many upstream targets are up to date

### ISSUE THREAD

**Issue by kendonB** [ropensci/drake#448]
Finally trying out "future" parallelism. I have 3 targets each `evaluate_plan`'ed into 10000 pieces.

The main [`while` loop](https://github.com/ropensci/drake/blob/5de5d088956f86cbf2e89fa83b3ab2f13c08e0c4/R/future.R#L5) takes a really long time to get through the first 20000 up to date targets before starting on target 20001. I believe this work is done using `lightly_parallelize` in other types of parallelism so it's not so bad.

Perhaps this method could trim the queue first by subsetting using the `outdated` function?

Of course, this is another issue that'll be easier after dealing with #440.

**Comment by wlandau** [comment]
#440 may speed this up to a degree, and there may be ways to locally parallelize parts of the master process. I will think about it. But for you, maybe a different tack is more appropriate. You have a ton of targets, but the structure of the dependency network is extremely simple. I think I could help you far more by adding a new `"clustermq_staged"` backend (described at https://github.com/mschubert/clustermq/issues/86#issuecomment-401358553) and a `"future_lapply_staged"` backend. I do not usually recommend [staged parallelism](https://deploy-preview-184--ropensci.netlify.com/technotes/2018/05/18/drake-hpc/#staged-scheduling), but neither do I believe that the perfect should be the enemy of the good.
